Clarify the definition of "inside conditions" (#3542)
reactjs/reactjs.org: content/docs/hooks-rules.md

### Only Call Hooks at the Top Level {#only-call-hooks-at-the-top-level}

> **Don't call Hooks inside loops, conditions, or nested functions.**. Instead, always use Hooks at the top level of your React function, before any early returns. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That's what allows React to correctly preserve the state of Hooks between multiple `useState` and `useEffect` calls. (If you're curious, we'll explain this in depth [below](#explanation).)

Author: Whoops, left an extra period...

```suggestion
**Don't call Hooks inside loops, conditions, or nested functions.** Instead, always use Hooks at the top level of your React function, before any early returns. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That's what allows React to correctly preserve the state of Hooks between multiple `useState` and `useEffect` calls. (If you're curious, we'll explain this in depth [below](#explanation).)
```

> **Don't call Hooks inside loops, nested functions, or conditions** (this includes calling them after "early exit" conditional return statements). Instead, always use Hooks at the top level of your React function. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That's what allows React to correctly preserve the state of Hooks between multiple `useState` and `useEffect` calls. (If you're curious, we'll explain this in depth [below](#explanation).)

Reviewer: What do you think about:

```suggestion
**Don't call Hooks inside loops, conditions, or nested functions.**. Instead, always use Hooks at the top level of your React function, before any early returns. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That's what allows React to correctly preserve the state of Hooks between multiple `useState` and `useEffect` calls. (If you're curious, we'll explain this in depth [below](#explanation).)
```

Author: Ooh, I like it!
